### DISCUSSION TITLE
IMPORTANT: Repository prune of bc-java

### DISCUSSION THREAD

**Discussion by dghgit** [bcgit/bc-java#1351] [Announcements]
Hi, we've removed all the PQC test vectors from bc-java (after making a back up). While this has lead to a dramatic reduction in repository size, the "commit adjustment" phase of the trim has meant everyone will have to re-sync (getting rid of the files changes the number of commits...)

I'm not sure how the github mirroring will react to this (my nervous wait has just begun, it should just overwrite, I think the problems will be on forks), nor how clones built on the mirror will. If it reacts the same way as it did for me there'll be complaints in clones about divergent commits. If you just reclone that will do with it. It's been 10 years of commits so far, definitely time a spring clean, we'll try and make sure it's a good ten years before the next one. Sorry. We're hoping by adding the new bc-test-data repository we can keep this kind of thing out of the source repository altogether.

With the test data - there is now a separate repository bc-test-data which has most of the library test data in it. If this repo is checked out either next to bc-java or at the top level of bc-java you should find the tests pick up the resource files they need.

If anyone finds themselves in a total mess over this, leave a message here and we will try to help. Gt will refuse to merge the divergences so local history should be okay, but there will be a period of adjustment.

In answer to the question of why, you'll see why when you do the new clone.

Thanks,

David

**Comment by vanitasvitae** [comment]
Just noticed when I had to manually re-apply commits in my PRs when suddenly there were thousands of commits of difference reported :D

**Reply by dghgit** [reply]
Yes, sorry... it's dammed if you do, dammed if you don't. On the updside the source repository is a fraction of the size it used to be and any further activity like this will be kept to bc-test-data.
